xds/src/main/java/io/grpc/xds/XdsLrsClient.java

```diff
@@ -0,0 +1,346 @@
+      long interval = reportStopwatch.elapsed(TimeUnit.NANOSECONDS);
+      reportStopwatch.reset().start();
+      lrsRequestWriter.onNext(LoadStatsRequest.newBuilder()
+          .setNode(Node.newBuilder()
+              .setMetadata(Struct.newBuilder()
+                  .putFields(
+                      TRAFFICDIRECTOR_HOSTNAME_FIELD,
+                      Value.newBuilder().setStringValue(serviceName).build())))
+          .addClusterStats(loadReportStore.generateLoadReport(Durations.fromNanos(interval)))
+          .build());
+      scheduleNextLoadReport();
+    }
+
+    private void scheduleNextLoadReport() {
+      // Cancel pending load report and reschedule with updated load reporting interval.
+      if (loadReportTimer != null && loadReportTimer.isPending()) {
+        loadReportTimer.cancel();
+        loadReportTimer = null;
+      }
+      if (loadReportIntervalNano > 0) {
+        loadReportTimer = syncContext.schedule(
+            new LoadReportingTask(this), loadReportIntervalNano, TimeUnit.NANOSECONDS,
+            timerService);
+      }
+    }
+
+    private void handleResponse(LoadStatsResponse response) {
+      if (closed) {
+        return;
+      }
+      logger.log(ChannelLogLevel.DEBUG, "Got an LRS response: {0}", response);
+
+      if (!initialResponseReceived) {
+        initialResponseReceived = true;
+      }
+      loadReportIntervalNano = Durations.toNanos(response.getLoadReportingInterval());
+      List<String> serviceList = Collections.unmodifiableList(response.getClustersList());
+      // For gRPC use case, LRS response will only contain one cluster, which is the same as in
+      // the EDS response.
+      if (serviceList.size() != 1 || !serviceList.get(0).equals(serviceName)) {
+        logger.log(ChannelLogLevel.ERROR, "Unmatched cluster name(s): %s with EDS response: %s",
+            serviceList, serviceName);
+        return;
+      }
+      scheduleNextLoadReport();
+    }
+
+    private void handleStreamClosed(Status status) {
+      checkArgument(!status.isOk(), "unexpected OK status");
+      if (closed) {
+        return;
+      }
+      closed = true;
+      cleanUp();
+
+      long delayNanos = 0;
+      if (initialResponseReceived || lrsRpcRetryPolicy == null) {
+        // Reset the backoff sequence if balancer has sent the initial response, or backoff sequence
+        // has never been initialized.
+        lrsRpcRetryPolicy = backoffPolicyProvider.get();
+      }
+      // Backoff only when balancer wasn't working previously.
+      if (!initialResponseReceived) {
+        // The back-off policy determines the interval between consecutive RPC upstarts, thus the
+        // actual delay may be smaller than the value from the back-off policy, or even negative,
+        // depending how much time was spent in the previous RPC.
+        delayNanos =
+            lrsRpcRetryPolicy.nextBackoffNanos() - retryStopwatch.elapsed(TimeUnit.NANOSECONDS);
+      }
+      if (delayNanos <= 0) {
```

Comment:
You probably want to log a DEBUG log here with the delayNanos. Also think about other places you may find DEBUG logs useful when debugging this class in a production environment.

Reply:
Added some. How does it look now?